src/portal_visualization/builders/epic_builders.py

```diff
@@ -43,17 +53,102 @@ def apply(self):
     def _apply(self, conf):  # pragma: no cover
         pass
 
+    def zarr_store_url(self):
+        adata_url = self._build_assets_url(zarr_path, use_token=False)
+        return adata_url
+
+    def segmentations_url(self, img_path):
+        img_url = self._build_assets_url(img_path)
+        return (
+            img_url,
+            str(
+                re.sub(
+                    r"ome\.tiff?",
+                    "offsets.json",
+                    re.sub(IMAGE_PYRAMID_DIR, OFFSETS_DIR, img_url),
+                )
+            ),
+        )
 
-class SegmentationMaskBuilder(EPICConfBuilder):
 
+class SegmentationMaskBuilder(EPICConfBuilder):  # pragma: no cover
     def _apply(self, conf):
+        zarr_url = self.zarr_store_url()
         datasets = conf.get_datasets()
-        print(f"Found {len(datasets)} datasets")
-        # Proof of concept using one of the kaggle segmentation masks for now
-        # segmentations = ObsSegmentationsOmeTiffWrapper(
-        #     img_url='https://assets.hubmapconsortium.org/c9d9ab5c9ee9642b60dd351024968627/ometiff-pyramids/VAN0042-RK-3-18-registered-PAS-to-postAF-registered.ome_mask.ome.tif?token=AgndN7NVbn83wwDXjpnY1Y0lDoJj2j7zOGmn1WN6qr9pqdkjKmt9C1XYm4KrlWrOXE9rVJvpnEKrPjIXrlKd1hmDGjV',
-        #     # offsets_path=f'./{name}/{name}/offsets/{name}.segmentations.offsets.json',
-        #     obs_types_from_channel_names=True,
-        # )
-        # dataset.add_object(segmentations)
-        pass
+        # TODO: add the correct path to the segmentation mask ome-tiff (image-pyramid)
+        seg_path = f'{self.segmentations_url("seg")}/'
+        # print(seg_path)
+        seg_path = (
+            'https://assets.hubmapconsortium.org/c9d9ab5c9ee9642b60dd351024968627/'
+            'ometiff-pyramids/VAN0042-RK-3-18-registered-PAS-to-postAF-registered.ome_mask.ome.tif?'
+            'token=AgzQXm7nvOW32vWw0EPpKonwbOqjNBzNvvW1p15855NoYglJxyfkC8rlJJWy8V6E8MeyXOwlpKdNBnHb5qnv7f8oeeG'
+        )
+        mask_names = self.read_metadata_from_url()
+        mask_names = ['mask1', 'mask2']  # for testing purposes
+        if (mask_names is not None):
+            segmentation_objects = create_segmentation_objects(zarr_url, mask_names)
+            segmentations = ObsSegmentationsOmeTiffWrapper(
+                img_url=seg_path,
+                obs_types_from_channel_names=True,
+                coordination_values={
+                    "fileUid": "segmentation-mask"
+                }
+            )
+
+            for dataset in datasets:
+                dataset.add_object(segmentations)
+                for obj in segmentation_objects:
+                    dataset.add_object(obj)
+
+                # TODO: what happens if these views already exist , and if there are other views, how to place these?
```

Comment:
Good question, this is definitely important to determine! While we may be able to handle the more basic image-only assays like Histology/PAS microscopy by appending to existing configurations, given that we anticipate there will be segmentation masks for various assays and that we'll have a few different base confs as a result, my understanding is that we may need to re-set up views and coordinations from scratch for those. We should discuss with @keller-mark to confirm/see if he has any alternative suggestions.